Oswaldo Vazquez Aparicio/30 Agosto/hackerrank/JavaArrayList.java

```diff
@@ -0,0 +1,75 @@
+package hackerrank;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public JavaArrayList {
+
+    private static final Scanner scanner = new Scanner(System.in);
+    private static List<List<Integer>> table;
```

Comment:
c'mon!!! 
You changed ArrayList<ArrayList<Integer>> for List<List<Integer>>
The thing is: why List<Integer> is not sufficient?

Reply:
I changed to List instead of ArrayList because I read in the resource below that you shared, it said that is better to use the Interface, I thought that was my mistake hehe.

https://www.javaguides.net/2018/06/java-collection-framework-best-practices.html

I did my matryoshka of ArrayLists because I wanted to do something like the image on the exercise in HackerRank, in this case, I imagined the red rectangle like an ArrayList that would contain
other rectangles and these rectangles would contain a number, the exercise was about ArrayList so I decided to use an ArrayList instead of an Array, so I ended with ArrayList<ArrayList<Integer>>
then later I changed to List<List<Integer>> thinking that was the problem, Oops!.

![1489168616-b25dd38013-arraylist](https://user-images.githubusercontent.com/47643347/131751971-0d30691c-e482-4209-9f6e-94803790a8df.png)

So basically that is how I come with this.

Now if the exercise wasn't about ArrayList, I could have used a two-dimension Array something like Integer[][] but the rows needed to be dynamic, so an Array wouldn't do the trick, so I think I could use an Array of ArrayList, in this case, something like this "ArrayList[] tableListArray", should I change it? Thanks for the feedback.

Reply:
OK